x/superfluid/keeper/intermediary_account.go

```diff
@@ -113,11 +113,11 @@ func (k Keeper) DeleteIntermediaryAccount(ctx sdk.Context, address sdk.AccAddres
 	prefixStore.Delete(address)
 }
 
-func (k Keeper) SetLockIdIntermediaryAccountConnection(ctx sdk.Context, lockId uint64, acc types.SuperfluidIntermediaryAccount) {
+func (k Keeper) SetLockIdIntermediaryAccountConnection(ctx sdk.Context, lockId uint64, acc sdk.AccAddress) {
```

Comment:
Can we keep it as before, I prefer the type safety it implied. (And instead make the genesis logic do the validation to get the superfluid intermediary account)